src/core/server.cc

```diff
@@ -188,9 +188,9 @@ InferenceServer::Init()
 }
 
 Status
-InferenceServer::Stop()
+InferenceServer::Stop(const bool force)
 {
-  if (ready_state_ != ServerReadyState::SERVER_READY) {
+  if (!force && (ready_state_ != ServerReadyState::SERVER_READY)) {
     return Status::Success;
```

Comment:
So returning early from here was the root cause, right?

Reply:
The problem was returning early from the Init() function without waiting for any in-flight threads to complete (which is what Stop() does).